pkg/odo/cli/component/create.go

```diff
@@ -942,109 +936,137 @@ func (co *CreateOptions) downloadStarterProject(projectPassed string, interactiv
 		return err
 	}
 
-	var logUrl, url, sparseDir string
-	if project.Git != nil {
-		if strings.Contains(project.Git.Location, "github.com") {
-			url, err = util.GetGitHubZipURL(project.Git.Location, project.Git.Branch, project.Git.StartPoint)
-			if err != nil {
-				return err
-			}
-			sparseDir = project.Git.SparseCheckoutDir
+	log.Info("\nStarter Project")
+
+	if starterProject.Git != nil || starterProject.Github != nil {
+
+		projectSource := common.GitLikeProjectSource{}
+		if starterProject.Git != nil {
+			projectSource = starterProject.Git.GitLikeProjectSource
 		} else {
-			return errors.Errorf("project type git with non github url not supported")
+			projectSource = starterProject.Github.GitLikeProjectSource
+		}
+
+		remoteName, remoteUrl, reference, err := projectSource.GetDefaultSource()
+		if err != nil {
+			return errors.Wrapf(err, "unable to get default project source for starter project %s", starterProject.Name)
+		}
+
+		downloadSpinner := log.Spinnerf("Downloading starter project %s from %s", starterProject.Name, remoteUrl)
+
+		_, err = git.PlainClone(path, false, &git.CloneOptions{
+			URL:           remoteUrl,
+			RemoteName:    remoteName,
+			ReferenceName: plumbing.ReferenceName(reference),
```

Comment:
not sure if am doing something wrong here,  the descriptions says if `revision` not found download default one, It is giving error if specified revision does not exist.
https://github.com/openshift/odo/pull/3867/files#diff-5e7d52b6df7f967077e7b6ab084ca107R186

I tried specifying  valid commit id, branch name, giving error
```
 ✗  couldn't find remote ref "refs/heads/450235d421864f9a1a47a06788be69a321ad63de"
```

am i missing the correct format to specify revisions here?

Reply:
To be honest I don't know :-( 
The reference is currently not working. The problem is that ideally, we should know if it is branch, tag, or sha.
I'm thinking about marking "reference" as unsupported for now and try to deal with it in followup PR. What do you think?

Reply:
Agreed with it, may be we log error or show warning that reference is not supported.

Reply:
Added warning, when revision is set.
If someone figures out how to pass a revision so it is accepted by go-git they will be able to use it and still see the warning.


```
Starter Project
 ⚠  Specifying 'revision' in 'checkoutFrom' is not yet supported in odo.
 ✗  Downloading starter project nodejs-starter from https://github.com/odo-devfiles/nodejs-ex.git [466ms]
 ✗  couldn't find remote ref "refs/heads/e832173192434d38d7a5380152e500c8de96b99e"
```